### PR TITLE
Query Builder Component

### LINKED ISSUE #11750: angular-query-builder  (Angular Tree Grid Advanced Filtering without the tree grid)
## Is your feature request related to a problem? Please describe.
We need to query/filter elements that are not from the grid but from a 3D viewer that we use or other components.

## Describe the solution you'd like
Use the "Angular Tree Grid Advanced Filtering" independently of the Tree Grid.

## Describe alternatives you've considered
Currently we are using https://www.syncfusion.com/angular-ui-components/angular-query-builder

## Additional context
We need the same component of the Angular Tree Grid Advanced Filtering but that works independently of the grid. And please add a search to find parameters while doing the conditions.

Thanks!
Antonio

### PR DESCRIPTION
**Pull request by mtsvyatkova**
Closes #11750 

### Additional information (check all that apply):
 - [ ] Bug fix
 - [X] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [ ] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/query-builder/query-builder.component.ts [hunk 1]
```diff
@@ -0,0 +1,1270 @@
...
+    constructor(public cdr: ChangeDetectorRef,
+        protected iconService: IgxIconService,
+        protected platform: PlatformUtil,
+        @Inject(LOCALE_ID) protected _localeId: string,
+        @Optional() @Inject(DisplayDensityToken) protected _displayDensityOptions?: IDisplayDensityOptions) {
+        super(_displayDensityOptions);
+        this.locale = this.locale || this._localeId;
+    }
+
+    public ngAfterViewInit(): void {
```
Comment (ChronosSF) [inline projects/igniteui-angular/src/lib/query-builder/query-builder.component.ts:354]: this should also be hidden internal

**Review by ChronosSF — CHANGES_REQUESTED** [review]
Please add the following:
- README about the component in the component's folder
- Modify the overall readme to include this component in the component grid
- Changelog entry

**Comment by didimmova** [comment]
@simeonoff 

Screenshot 2022-09-28 at 14 39 15
- Buttons are not aligned with the fields

Screenshot 2022-09-28 at 14 41 06
- The icon in the igx-prefix also seems to be misaligned

**Comment by simeonoff** [comment]
> @simeonoff
> 
> Screenshot 2022-09-28 at 14 39 15
> 
> -Buttons are not aligned with the fields
> Screenshot 2022-09-28 at 14 41 06
> 
> -The icon in the igx-prefix also seems to be misaligned

These must be preexisting issues, as I've not done any modifications to the theme that could introduce them. I suggest we log them as separate issues and fix in another PR so that this one doesn't get dragged any further.

**Review by didimmova — DISMISSED** [review]

**Review by ChronosSF — APPROVED** [review]

**Pull request merged**